enhancements/project-request-limit.md

```diff
@@ -0,0 +1,345 @@
+* We treat the `ProjectRequestLimit` admission webhook as a cluster singleton.
+* The CRD will be `cluster-scoped`.
+* The operator will be reconciling a CR named `cluster`, it will ignore other custom resources.
+
+```
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: projectrequestlimits.operator.project.openshift.io
+spec:
+  group: operator.project.openshift.io
+  version: v1
+  scope: Cluster
+  names:
+    plural: projectrequestlimits
+    singular: projectrequestlimit
+    kind: ProjectRequestLimit
+```  
+
+This is how a cluster-admin can enable the `ProjectRequestLimit` admission webhook on a cluster with `limits` to be enforced. 
+```
+apiVersion: operator.project.openshift.io/v1
+kind: ProjectRequestLimitConfig
+metadata:
+  name: cluster
+spec:
+  limits:
+  - selector:
+      level: admin
+  - selector:
+      level: advanced
+    maxProjects: 10
+  - maxProjects: 2
+```
+
+##### Install:
+* The operator is installed into a predefined namespace by OLM: The predefined namespace `openshift-project-request-limit-operator` 
+  will be wired in to the OLM manifests.
+* The operator will install the `ProjectRequestLimit` admission webhook into the same namespace as the operator. 
+
+
+##### Configuration:
+The operand is immutable, it loads the configuration from a file when it starts. A `ConfigMap` containing the configuration
+is mounted as a volume into the Pod at a certain file path.
+```
+apiVersion: apps/v1
+kind: DaemonSet
+spec:
+  template:
+    metadata:
+      annotations:
+        projectrequestlimits.operator.project.openshift.io/configuration.hash: abcdefgh
+    spec:
+      containers:
+        - name: project-request-limit-admission
+          command:
+            - /usr/bin/project-request-limit-admission
+          args:
+            - "--configuration=$(CONFIGURATION_PATH)"
+          env:
+            - name: CONFIGURATION_PATH
+              value: /etc/projectrequestlimit/config/limits.yaml
+          volumeMounts:
+            name: configuration
+            mountPath: /etc/projectrequestlimit/config/limits.yaml
+            subPath: limits.yaml
+      volumes:
+        - name: configuration
+          configMap:
+            name: project-request-limit-configuration
+``` 
+
+The operator will ensure that change(s) made to the configuration in the `cluster` custom resource are propagated to the 
+`ProjectRequestLimit` admission webhook immediately.
+* The cluster admin changes the configuration inside the `cluster` custom resource.
+* The operator detects the change and compares the hash of the `desired` configuration with that of the `current` configuration.
+* If the configuration has changed, the operator updates the `ConfigMap` object and will set the annotation 
+  `projectrequestlimits.operator.project.openshift.io/configuration.hash` of the template `PodSpec` of the operand.
```

Comment:
I don't see why we need this indirection with a configmap. Just use the operator config directly.

Reply:
we need a ConfigMap to volume mount the configuration in to the Pod, is there any other way to mount configuration into a Pod directly from a custom resource?